x-pack/plugins/apm/public/context/time_range_metadata/time_range_metadata_context.tsx

```diff
@@ -46,6 +47,38 @@ export function TimeRangeMetadataContextProvider({
 
   const routePath = useApmRoutePath();
 
+  const isOperationView =
+    routePath.startsWith('/dependencies/operation') ||
+    routePath.startsWith('/dependencies/operations');
+
+  return (
+    <TimeRangeMetadataContextProvider
+      uiSettings={uiSettings}
+      useSpanName={isOperationView}
```

Comment:
Something still feels odd about this option (`useSpanName`). I don't understand why we have to special-case service destination metrics like this - isn't it just another source similar to transaction metrics and transaction events?

I think the response of `GET /internal/apm/time_range_metadata` shows the inconsistency:

```json
{
  "isUsingServiceDestinationMetrics": true,
  "sources": [
    {
      "documentType": "serviceTransactionMetric",
      "rollupInterval": "1m",
      "hasDocs": true
    },
    {
      "documentType": "serviceTransactionMetric",
      "rollupInterval": "10m",
      "hasDocs": true
    },
    {
      "documentType": "serviceTransactionMetric",
      "rollupInterval": "60m",
      "hasDocs": true
    },
    {
      "documentType": "transactionMetric",
      "rollupInterval": "1m",
      "hasDocs": true
    },
    {
      "documentType": "transactionMetric",
      "rollupInterval": "10m",
      "hasDocs": true
    },
    {
      "documentType": "transactionMetric",
      "rollupInterval": "60m",
      "hasDocs": true
    },
    {
      "documentType": "transactionEvent",
      "rollupInterval": "none",
      "hasDocs": true
    }
  ]
}

```

Shouldn't the service destination metrics be part of sources, like:

```json
{
  "sources": [
    {
      "documentType": "serviceDestinationMetrics",
      "rollupInterval": "none",
      "hasDocs": true
    },
    {
      "documentType": "spanEvents",
      "rollupInterval": "none",
      "hasDocs": true
    },
    // ...
  ]
}
```

Apart from making things more consistent, we're now able to have views where one component uses `serviceDestinationMetrics` and another uses `spanEvents` similar to what we do with transactions.

Reply:
In earlier versions `span.name` was not available on span metrics. That was a fairly recent change. But we have a couple of views that were added after, and require `span.name` to be present. If `span.name` is not available on span metrics, we fall back to span events.

Reply:
Then what about something like this?

```
{
  "sources": [
    {
      "documentType": "spanMetricsWithName",
      "rollupInterval": "none",
      "hasDocs": false
    },
    {
      "documentType": "spanMetrics",
      "rollupInterval": "none",
      "hasDocs": true
    },
    {
      "documentType": "spanEvents",
      "rollupInterval": "none",
      "hasDocs": true
    },
    // ...
  ]
}
```

Reply:
I've considered that, also in the context of transaction metrics with `transaction.duration.summary`, but I don't think different document types for schema changes will scale. I would rather return some metadata with the document type, e.g. a version or a `meta` property. But I will tackle that when I implement support for `transaction.duration.summary` on tx metrics (which might not include changes to service destination metrics as it makes sense to tackle that separately).